src/main/java/spoon/MavenLauncher.java

```diff
@@ -108,656 +125,144 @@ public MavenLauncher(String mavenProject, String m2RepositoryPath, SOURCE_TYPE s
-						version = v.version;
-						break;
-					}
-				}
-			}
-			return version;
-		}
-
-		private InheritanceModel readPom(String groupId, String artifactId, String version) {
-			version = extractVersion(groupId, artifactId, version);
-			groupId = groupId.replace(".", "/");
-			String fileName = artifactId + "-" + version;
-			Path depPath = Paths.get(m2RepositoryPath, groupId, artifactId, version, fileName + ".pom");
-			try {
-				return readPOM(depPath.toString(), null);
-			} catch (Exception e) {
-				LOGGER.log(Level.ERROR, "Unable to read the POM:" + depPath.toString(), e);
-				return null;
-			}
 		}
+		String[] classpath = new String[classpathElements.size()];
+		classpath = classpathElements.toArray(classpath);
+		return classpath;
+	}
 
-		private TreeDependency getTreeDependency(Dependency dependency, boolean isLib, Set<TreeDependency> hierarchy) {
-			String groupId = extractVariable(dependency.getGroupId());
-			String artifactId = extractVariable(dependency.getArtifactId());
-			String version = extractVersion(groupId, artifactId, dependency.getVersion());
-			if (version == null) {
-				LOGGER.warn("A dependency version cannot be resolved: " + groupId + ":" + artifactId + ":" + version);
-				return null;
-			}
-			// pass only the optional dependency if it's in a library dependency
-			if (isLib && dependency.isOptional()) {
-				return null;
-			}
-
-			// ignore test dependencies for app source code
-			if ("test".equals(dependency.getScope()) && (SOURCE_TYPE.APP_SOURCE == sourceType || isLib)) {
-				return null;
-			}
-			// ignore not transitive dependencies
-			if (isLib && ("test".equals(dependency.getScope()) || "provided".equals(dependency.getScope()) || "compile".equals(dependency.getScope()))) {
-				LOGGER.log(Level.WARN, "Dependency ignored (scope: provided or test):" + dependency.getGroupId() + ":" + dependency.getArtifactId() + ":" + version);
-				return null;
-			}
-			TreeDependency dependence = new TreeDependency(groupId, artifactId, version, dependency.getType());
-			try {
-				InheritanceModel dependencyModel = readPom(groupId, artifactId, version);
-				if (dependencyModel != null) {
-					dependence = dependencyModel.getTreeDependency(true, hierarchy);
-					dependence.groupId = groupId;
-					dependence.artifactId = artifactId;
-					dependence.version = version;
+	static String guessMavenHome() {
+		String mvnHome = null;
+		try {
+			String[] cmd = {"mvn", "-version"};
+			Process p = Runtime.getRuntime().exec(cmd);
+			BufferedReader output = new BufferedReader(new InputStreamReader(p.getInputStream()));
+			String line;
 
-					if (dependency.getExclusions() != null) {
-						for (int i = 0; i < dependency.getExclusions().size(); i++) {
-							Exclusion exclusion = dependency.getExclusions().get(i);
-							dependence.removeDependency(exclusion.getGroupId(), exclusion.getArtifactId());
-						}
-					}
+			while ((line = output.readLine()) != null) {
+				if (line.contains("Maven home: ")) {
```

Comment:
Could you use a constant for "Maven home: " please?

Reply:
Yes, indeed.